### ISSUE TITLE
Ubuntu20.04: ERRO Error starting stopped VM: virError(Code=55, Domain=18, Message='Requested operation is not valid: format of backing image '/home/ss97/.crc/cache/crc_libvirt_4.5.14/crc.qcow2' of image '/home/ss97/.crc/machines/crc/crc.qcow2' was not specified in the image metadata (See https://libvirt.org/kbase/backing_chains.html for troubleshooting)')

### ISSUE THREAD

**Issue by vyom-soft** [code-ready/crc#1596]
### General information

  * OS: Linux / (Ubuntu20.04)
  * Hypervisor:Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? Yes

## CRC version 1.17.0
```
  
## CRC status
CRC VM:          Stopped
OpenShift:       Stopped
Disk Usage:      0B of 0B (Inside the CRC VM)
Cache Usage:     13.04GB
Cache Directory: /home/ss97/.crc/cache


## CRC config
```bash
# Put the output of `crc config view`
```

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
```

### Logs

You can start crc with `crc start --log-level debug` to collect logs.
<pre><font color="#D3D7CF">DEBU</font> CodeReady Containers version: 1.17.0+99f5c87 
<font color="#D3D7CF">DEBU</font> OpenShift version: 4.5.14 (embedded in binary) 
<font color="#D3D7CF">DEBU</font> Total memory of system is 12508778496 bytes  
<font color="#D3D7CF">DEBU</font> Unable to find out if a new version is available: Get https://mirror.openshift.com/pub/openshift-v4/clients/crc/latest/release-info.json: dial tcp: lookup mirror.openshift.com: no such host 
<font color="#06989A">INFO</font> Checking if oc binary is cached              
<font color="#D3D7CF">DEBU</font> Running &apos;/home/ss97/.crc/bin/oc/oc version --client&apos; 
<font color="#D3D7CF">DEBU</font> Found oc version 4.5.14                      
<font color="#D3D7CF">DEBU</font> oc binary already cached                     
<font color="#06989A">INFO</font> Checking if podman remote binary is cached   
<font color="#D3D7CF">DEBU</font> Currently podman remote is not supported     
<font color="#06989A">INFO</font> Checking if goodhosts binary is cached       
<font color="#D3D7CF">DEBU</font> goodhost binary already cached               
<font color="#06989A">INFO</font> Checking minimum RAM requirements            
<font color="#D3D7CF">DEBU</font> Total memory of system is 12508778496 bytes  
<font color="#06989A">INFO</font> Checking if running as non-root              
<font color="#06989A">INFO</font> Checking if Virtualization is enabled        
<font color="#D3D7CF">DEBU</font> Checking if the vmx/svm flags are present in /proc/cpuinfo 
<font color="#D3D7CF">DEBU</font> CPU virtualization flags are good            
<font color="#06989A">INFO</font> Checking if KVM is enabled                   
<font color="#D3D7CF">DEBU</font> Checking if /dev/kvm exists                  
<font color="#D3D7CF">DEBU</font> /dev/kvm was found                           
<font color="#06989A">INFO</font> Checking if libvirt is installed             
<font color="#D3D7CF">DEBU</font> Checking if &apos;virsh&apos; is available             
<font color="#D3D7CF">DEBU</font> &apos;virsh&apos; was found in /usr/bin/virsh          
<font color="#06989A">INFO</font> Checking if user is part of libvirt group    
<font color="#D3D7CF">DEBU</font> Checking if current user is part of the libvirt group 
<font color="#D3D7CF">DEBU</font> Running &apos;/usr/bin/groups ss97&apos;               
<font color="#D3D7CF">DEBU</font> Current user is already in the libvirt group 
<font color="#06989A">INFO</font> Checking if libvirt daemon is running        
<font color="#D3D7CF">DEBU</font> Checking if libvirtd service is running      
<font color="#D3D7CF">DEBU</font> Running &apos;systemctl status virtqemud.socket&apos;  
<font color="#D3D7CF">DEBU</font> Command failed: exit status 4                
<font color="#D3D7CF">DEBU</font> stdout:                                      
<font color="#D3D7CF">DEBU</font> stderr: Unit virtqemud.socket could not be found. 
<font color="#D3D7CF">DEBU</font> virtqemud.socket is not running              
<font color="#D3D7CF">DEBU</font> Running &apos;systemctl status libvirtd.socket&apos;   
<font color="#D3D7CF">DEBU</font> libvirtd.socket is running                   
<font color="#06989A">INFO</font> Checking if a supported libvirt version is installed 
<font color="#D3D7CF">DEBU</font> Checking if libvirt version is &gt;=3.4.0       
<font color="#D3D7CF">DEBU</font> Running &apos;virsh -v&apos;                           
<font color="#06989A">INFO</font> Checking if crc-driver-libvirt is installed  
<font color="#D3D7CF">DEBU</font> Checking if crc-driver-libvirt is installed  
<font color="#D3D7CF">DEBU</font> Running &apos;/home/ss97/.crc/bin/crc-driver-libvirt version&apos; 
<font color="#D3D7CF">DEBU</font> Found crc-driver-libvirt version 0.12.10     
<font color="#D3D7CF">DEBU</font> crc-driver-libvirt is already installed      
<font color="#06989A">INFO</font> Checking if libvirt &apos;crc&apos; network is available 
<font color="#D3D7CF">DEBU</font> Checking if libvirt &apos;crc&apos; network exists     
<font color="#D3D7CF">DEBU</font> Running &apos;virsh --connect qemu:///system net-info crc&apos; 
<font color="#D3D7CF">DEBU</font> Checking if libvirt &apos;crc&apos; definition is up to date 
<font color="#D3D7CF">DEBU</font> Running &apos;virsh --connect qemu:///system net-dumpxml --inactive crc&apos; 
<font color="#D3D7CF">DEBU</font> libvirt &apos;crc&apos; network has the expected value 
<font color="#06989A">INFO</font> Checking if libvirt &apos;crc&apos; network is active  
<font color="#D3D7CF">DEBU</font> Checking if libvirt &apos;crc&apos; network is active  
<font color="#D3D7CF">DEBU</font> Running &apos;virsh --connect qemu:///system net-info crc&apos; 
<font color="#D3D7CF">DEBU</font> libvirt &apos;crc&apos; network is already active      
<font color="#06989A">INFO</font> Checking if NetworkManager is installed      
<font color="#D3D7CF">DEBU</font> Checking if &apos;nmcli&apos; is available             
<font color="#D3D7CF">DEBU</font> &apos;nmcli&apos; was found in /usr/bin/nmcli          
<font color="#06989A">INFO</font> Checking if NetworkManager service is running 
<font color="#D3D7CF">DEBU</font> Checking if NetworkManager.service is running 
<font color="#D3D7CF">DEBU</font> Running &apos;systemctl status NetworkManager&apos;    
<font color="#D3D7CF">DEBU</font> NetworkManager.service is already running    
<font color="#06989A">INFO</font> Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists 
<font color="#D3D7CF">DEBU</font> Checking NetworkManager configuration        
<font color="#D3D7CF">DEBU</font> NetworkManager configuration is good         
<font color="#06989A">INFO</font> Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists 
<font color="#D3D7CF">DEBU</font> Checking dnsmasq configuration               
<font color="#D3D7CF">DEBU</font> dnsmasq configuration is good                
<font color="#06989A">INFO</font> Starting CodeReady Containers VM for OpenShift 4.5.14... 
<font color="#D3D7CF">DEBU</font> Updating CRC VM configuration                
<font color="#CC0000">ERRO</font> Error starting stopped VM: virError(Code=55, Domain=18, Message=&apos;Requested operation is not valid: format of backing image &apos;/home/ss97/.crc/cache/crc_libvirt_4.5.14/crc.qcow2&apos; of image &apos;/home/ss97/.crc/machines/crc/crc.qcow2&apos; was not specified in the image metadata (See https://libvirt.org/kbase/backing_chains.html for troubleshooting)&apos;) 
Error starting stopped VM: virError(Code=55, Domain=18, Message=&apos;Requested operation is not valid: format of backing image &apos;/home/ss97/.crc/cache/crc_libvirt_4.5.14/crc.qcow2&apos; of image &apos;/home/ss97/.crc/machines/crc/crc.qcow2&apos; was not specified in the image metadata (See https://libvirt.org/kbase/backing_chains.html for troubleshooting)&apos;)
</pre>

**Comment by guillaumerose** [comment]
Can you attach the crc.log file please (located in ~/.crc) ? I would like to know which operation is not valid.

**Comment by OlivierBondu** [comment]
Note: I am not the original poster, but I am running into the same issue (same OS, same version), the end of my crc.log is pasted below

```time="2020-10-15T14:24:07+02:00" level=info msg="Starting CodeReady Containers VM for OpenShift 4.5.14..."
time="2020-10-15T14:24:07+02:00" level=debug msg="Updating CRC VM configuration"
(crc) Calling .UpdateConfigRaw
(crc) Calling .UpdateConfigRaw
(crc) Calling .GetConfigRaw
(crc) Calling .Start
(crc) DBG | Starting VM crc
(crc) Failed to start: virError(Code=55, Domain=18, Message='Requested operation is not valid: format of backing image '/home/obondu/.crc/cache/crc_libvirt_4.5.14/crc.qcow2' of image '/home/obondu/.crc/machines/crc/crc.qcow2' was not specified in the image metadata (See https://libvirt.org/kbase/backing_chains.html for troubleshooting)')
time="2020-10-15T14:24:08+02:00" level=error msg="Error starting stopped VM: virError(Code=55, Domain=18, Message='Requested operation is not valid: format of backing image '/home/obondu/.crc/cache/crc_libvirt_4.5.14/crc.qcow2' of image '/home/obondu/.crc/machines/crc/crc.qcow2' was not specified in the image metadata (See https://libvirt.org/kbase/backing_chains.html for troubleshooting)')"
Making call to close driver server
(crc) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
```

**Comment by kchamit** [comment]
I am running ubuntu 20.4
After following all the manual setup steps at this [Link](https://github.com/code-ready/crc/issues/549) I faced the same issue.
This is what worked from me based on [this](https://libvirt.org/kbase/backing_chains.html) documentation.

Edit the VM image using
$ virsh edit crc

Locate the disk section and add following:
<backingStore type='file'>
    <format type='qcow2'/>
    <source file='/home/USER_NAME_HERE/.crc/cache/crc_libvirt_4.5.14/crc.qcow2'/>
    <backingStore/>
</backingStore>

Save and run. Works for me.

**Comment by guillaumerose** [comment]
Can you paste the output of `qemu-img info image.qcow2` for both images ? We might find something with this command.

Also, did you hit this bug ? https://github.com/code-ready/crc/issues/1578

**Comment by OlivierBondu** [comment]
Thanks a lot @kchamit ! I run into another issue but at least this one step seems solved.

@guillaumerose : outputs can be found below, and on my end no, I did not hit bug 1578
```
➜  ~ qemu-img info /home/obondu/.crc/cache/crc_libvirt_4.5.14/crc.qcow2
image: /home/obondu/.crc/cache/crc_libvirt_4.5.14/crc.qcow2
file format: qcow2
virtual size: 31 GiB (33285996544 bytes)
disk size: 9.97 GiB
cluster_size: 65536
Format specific information:
    compat: 1.1
    lazy refcounts: true
    refcount bits: 16
    corrupt: false
➜  ~ qemu-img info /home/obondu/.crc/machines/crc/crc.qcow2
image: /home/obondu/.crc/machines/crc/crc.qcow2
file format: qcow2
virtual size: 31 GiB (33285996544 bytes)
disk size: 53.1 MiB
cluster_size: 65536
backing file: /home/obondu/.crc/cache/crc_libvirt_4.5.14/crc.qcow2
Format specific information:
    compat: 1.1
    lazy refcounts: false
    refcount bits: 16
    corrupt: false
```

**Comment by guillaumerose** [comment]
backing file format is missing. According to the doc https://libvirt.org/kbase/backing_chains.html#vm-refuses-to-start-due-to-misconfigured-backing-store-format, you need to perform: 
```
qemu-img rebase -f qcow2 -F qcow2 -b /home/obondu/.crc/cache/crc_libvirt_4.5.14/crc.qcow2 /home/obondu/.crc/machines/crc/crc.qcow2
```

**Comment by OlivierBondu** [comment]
seems to work as well, except I am now running into the above mentioned issue #1578 ...

**Comment by guillaumerose** [comment]
Good to know. If you change the machine image to use the cache image, you will not be able to do crc delete/start and have a fresh install. You will have to delete the cache also.

**Comment by cfergeau** [comment]
Closing this issue as the fix for it has been merged in `machine-driver-libvirt`
